blackhole.c

```diff
@@ -547,17 +549,20 @@ blackhole_feedback_ngbiter(TreeWalkQueryBHFeedback * I,
         if(P[other].SwallowID != I->ID) return;
 
         lock_particle(other);
+
+        /* Mass conservation */
         O->Mass += (P[other].Mass);
         O->BH_Mass += (BHP(other).Mass);
+        P[other].Mass = 0;
+        BHP(other).Mass = 0;
```

Comment:
This changes the logic, because the Mass is zerod *before* it is accreted.

Reply:
good catch!